syntax/basic/function.vim

```diff
@@ -38,7 +38,7 @@ syntax match   typescriptArrowFuncDef          contained /\K\k*\s*=>/
   \ skipwhite skipempty
 
 " TODO: optimize this pattern
-syntax region   typescriptArrowFuncDef          contained start=/(\_[^)]*):/ end=/=>/
+syntax region   typescriptArrowFuncDef          contained start=/(\_[^(^)]*):/ end=/=>/
```

Comment:
I wonder if we need multiple `^` here?

Reply:
You're right, my mistake. This would mean that the start sequence will stop at the first caret, that is caret only have special meaning if it's the first character..